src/widget.cpp

```diff
@@ -910,6 +924,15 @@ window_panel widget::get_window_panel( const int width, const int req_height )
         for( const widget_id &wid : _widgets ) {
             height += wid->_height > 0 ? wid->_height : 1;
         }
+        for( const widget_clause &clause : _clauses ) {
+            int clause_height = 0;
+            for( const widget_id &wid : clause.widgets ) {
+                clause_height += wid->_height > 0 ? wid->_height : 1;
+            }
+            if( clause_height > height ) {
+                height = clause_height;
+            }
```

Comment:
Was hoping to reserve space for the clause with the largest height here, but it does not have the desired effect.

Currently, space can only be reserved by also defining dummy `widgets` for the layout, outside the clauses.

Reply:
Enabled W_DYNAMIC_HEIGHT for row-based layout widgets in 676d7c7.

Dynamic height and hiding empty widgets works now. If fixed height is desired, space can be reserved using the `height` property.